File: 240622/특정 문자로 끝나는 문자열/string-ending-with-specific-character.py
words = [input() for _ in range(10)]
find_chr = input()

answer = []
for word in words:
    if word.endswith(find_chr):
        answer.append(word)

if len(answer) == 0:
    print("None")
else:
    for word in answer:
        print(word)

File: 251130/특정 문자로 끝나는 문자열/string-ending-with-specific-character.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main() 
{
    vector<string> words(10);
    string find_chr;

    for (int i = 0; i < 10; i++) 
    {
        cin >> words[i];
    }

    cin >> find_chr;

    vector<string> answer;

    for (const string& word : words) 
    {
        if (word.size() >= find_chr.size() &&
            word.substr(word.size() - find_chr.size()) == find_chr) 
        {
            answer.push_back(word);
        }
    }

    if (answer.empty()) 
    {
        cout << "None";
    } 
    else 
    {
        for (const string& w : answer) 
        {
            cout << w << "\n";
        }
    }

    return 0;
}
